### PR TITLE
feat(autoware_launch): add e2e_simulator.launch.xml

### PR DESCRIPTION
**Pull request by shmpwk**
Signed-off-by: Shumpei Wakabayashi <shumpei.wakabayashi@tier4.jp>

## Description

Add the [degital twin simulation](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/digital-twin-simulation/awsim-tutorial/) launcher.
This launcher is temporal one. In the future, the simulator can be launched with `autoware.launch.xml`.

## Related links

- [TIERIV INTERNAL LINK](https://tier4.atlassian.net/browse/AJD-484)
- https://github.com/autowarefoundation/autoware.universe/pull/1991

## Tests performed

I tested this launcher with TIERIV INTERNAL DEGITAL TWIN SIMULATOR and it works as we expected.

## Notes for reviewers

What I want reviewers to check is
- the launcher name
  - `e2e_simulator.launch`, `degital_twin_simulator.launch` and etc..
- `enable_fine_detection` off 
  - The current trained model of fine detection is finetuned with the real environment. Thus it is difficult to adapt the simulation environment now. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/launch/autoware.launch.xml [hunk 1]
```diff
@@ -99,6 +101,7 @@
       <arg name="vehicle_param_file" value="$(find-pkg-share $(var vehicle_model)_description)/config/vehicle_info.param.yaml"/>
       <arg name="use_pointcloud_container" value="$(var use_pointcloud_container)"/>
       <arg name="pointcloud_container_name" value="$(var pointcloud_container_name)"/>
+      <arg name="enable_fine_detection" value="$(var enable_traffic_light_fine_detection)"/>
```
Comment (shmpwk) [inline autoware_launch/launch/autoware.launch.xml:104]: In [universe launcher](https://github.com/autowarefoundation/autoware.universe/pull/1991), the traffic light fine detection option is called `enable_fine_detection`.

File: autoware_launch/launch/e2e_simulator.launch.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      <arg name="launch_sensing" value="$(var sensing)"/>
+      <arg name="launch_localization" value="$(var localization)"/>
+      <arg name="launch_perception" value="$(var perception)"/>
+      <arg name="launch_planning" value="$(var planning)"/>
+      <arg name="launch_control" value="$(var control)"/>
+      <!-- Global parameters -->
+      <arg name="use_sim_time" value="true"/>
+      <!-- Pointcloud container -->
+      <arg name="use_pointcloud_container" value="$(var use_pointcloud_container)"/>
+      <arg name="pointcloud_container_name" value="$(var pointcloud_container_name)"/>
```
Comment (yukkysaito) [inline autoware_launch/launch/e2e_simulator.launch.xml:48]: I think it is better to remove to reduce the difference between autoware.launch and e2e.launch

File: autoware_launch/launch/e2e_simulator.launch.xml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      <!-- Vehicle -->
+      <arg name="launch_vehicle_interface" value="$(var launch_vehicle_interface)"/>
+      <!-- System -->
+      <arg name="system_run_mode" value="online"/>
```
Comment (yukkysaito) [inline autoware_launch/launch/e2e_simulator.launch.xml:52]: same comment
https://github.com/autowarefoundation/autoware_launch/pull/98/files#r984165931

File: autoware_launch/launch/e2e_simulator.launch.xml (hunk 2 above)
Comment (shmpwk) [reply autoware_launch/launch/e2e_simulator.launch.xml]: Thank you. I removed them by 3d8af47df1b112340056e0a2d7c25dc4b8b3ebd7.

File: autoware_launch/launch/e2e_simulator.launch.xml (hunk 3 above)
Comment (shmpwk) [reply autoware_launch/launch/e2e_simulator.launch.xml]: Thank you. I removed them by 3d8af47df1b112340056e0a2d7c25dc4b8b3ebd7.

**Comment by shmpwk** [comment]
Which launcher name do you like, `e2e_simulator.launch` or `degital_twin_simulator.launch`?
I think the former one is shorter but not sure if the abbreviation will be understood.
The latter one is redundant but eary to be understood.

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**